test/support/warning_filter.rb

```diff
@@ -0,0 +1,53 @@
+require 'pathname'
+require 'set'
+
+module RegexpParserTest
+  # Stream filter for warnings
+  class WarningFilter
```

Comment:
I wouldn't say this is necessarily a filter, it's more like something that captures all warnings, and something else handles the filtering.

With that said, I don't see any reason why this object can't take a target to wrap, and a whitelist of allowed warnings.